src/archive.cpp

```diff
@@ -70,15 +70,22 @@ namespace zim
   {
     if (m_impl->hasFrontArticlesIndex()) {
       return m_impl->getFrontEntryCount().v;
-    } else if (m_impl->hasNewNamespaceScheme()) {
-      return m_impl->getNamespaceEntryCount('C').v;
     } else {
-      return m_impl->getNamespaceEntryCount('A').v;
+      try {
+        return countMimeType(
+          getMetadata("Counter"),
+          [](const std::string& mimetype) { return mimetype.find("text/html") == 0; }
+        );
+      } catch(const EntryNotFound& e) {
+        if (m_impl->hasNewNamespaceScheme()) {
+          return m_impl->getNamespaceEntryCount('C').v;
+        } else {
+          return m_impl->getNamespaceEntryCount('A').v;
+        }
```

Comment:
You can avoid excessive nesting here via

```c++
const char articleNS = m_impl->hasNewNamespaceScheme() ? 'C' : 'A';
```

Reply:
Done